### ISSUE TITLE
Simplify "Pronunciation Check" feature for Reviewer

### ISSUE THREAD

**Issue by promike-github** [ankidroid/Anki-Android#16185]
**Is your feature request related to a problem? Please describe.**
I appreciate the great work done on Ankidroid. However, the new "Check Pronunciation" feature feels a bit overwhelming. It takes up too much space on the display and slows down card progression due to the increased number of touches required.

**Describe the solution you'd like**
I would like a simplified mode in the settings that resembles the old pronunciation check feature. This mode should be more streamlined and efficient, making it easier to navigate through cards quickly.

**Describe alternatives you've considered**
A simplified mode within the settings seems to be the most straightforward solution.

**Additional context**
Thank you for the excellent work you do on Ankidroid!

----

**Cause**
* #13043

**Comment by david-allison** [comment]
@promike-github 

Could you quickly work through what your ideal process would look like, on a tap-by-tap basis to be sure we're minimising effort on your behalf

@criticalAY FYI

**Comment by promike-github** [comment]
Thank you for your prompt response.

I was completely satisfied with the previous "Check pronunciation" feature. I'd appreciate it if this option could be made selectable within the menu system.

I have now downgraded to Anki because, for me, it was more efficient with the old one.
I am attaching a picture of the old feature for reference.
![Screenshot_20240415_163316_AnkiDroid](https://github.com/ankidroid/Anki-Android/assets/167091461/c72aefe3-d779-4408-9fdf-0f9613828004)

Thank you for considering this request.

**Comment by criticalAY** [comment]
> Of course, a tap and hold and a release automatically playing the recorded audio would be even more efficient, but I'm happy with the old one too.

I am not in favour of reverting the changes, though can alter the functionality as suggested minimising the steps. @david-allison I am planning if I can take it on during GSoC period (if selected/ or even if not selected) because I am going to peel of the old multimedia activity code anyway. I can separate the design for audio players in both Recorder and reviewer.

**Comment by david-allison** [comment]
@promike-github 
We're not going to go back directly to the old design, but we can make changes going forwards

What can we do with the new design to make things easier? Hiding the recording progress and turning it into an animated icon based on volume?

**Comment by promike-github** [comment]
@david-allison 
I totally understand the direction should be forward and not backwards.

Just so you know, I haven't been using the latest version much because the new design kind of slowed me down with all the extra taps it needed. Plus, it takes up quite a bit of space on the screen, especially when there's a lot of info or images to deal with.

I really liked how simple the old design was - record, stop, and play back in just three taps without having to "save" anything. (_I think a tap and hold feature combined with auto replay would be a game-changer in terms of efficiency._)

As for the idea of hiding the recording progress and using an animated volume-based icon, I'm not sure how much visual feedback really adds to the studying experience. It might just end up taking up space on the card without adding much value.

If you could personalize it in the settings a little bit, that would be fantastic, because I'm sure there are people who like it this way.

**Comment by criticalAY** [comment]
Tap hold -> records audio 
Release reorder button -> plays the audio + option to replay and reset 

This would remove few steps from the existing structure. Is this fine ?

**Comment by david-allison** [comment]
The idea of replacing the visualisation with an animated volume icon for the reviewer is to save space vertically.

I'm onboard with "press & hold"

@BrayanDSO would you see this functionality in Android or in HTML after the Reviewer redesign?

**Comment by criticalAY** [comment]
David the way we use the audio recorder is just by simply injecting the view to a layout we have in reviewer so I don't think we will have a problem here

**Comment by BrayanDSO** [comment]
> @BrayanDSO would you see this functionality in Android or in HTML after the Reviewer redesign?

I'm not proficient enough in Javascript to tell if all of that could be done with it, or if it is worth it. 

I don't use it personally, so I'll probably just go with the easiest approach or leave it to someone else when the time comes

**Comment by david-allison** [comment]
I personally wouldn't want audio to playback immediately, I'll consider either making it configurable, or seeing how we can make things work

For me, I can see uses for:

* immediate playback - if I'm on the back of the card, comparing my audio to sample audio
* playback immediately after a card is flipped: as above
* user-defined playback: A user would playback only if they think they got it incorrect

I think making the button multifunctional (hold/tap) is the right move

**Comment by criticalAY** [comment]
I will ping you for your input when I start working on it

**Comment by david-allison** [comment]
@promike-github This may take a little while to get off the ground, and we're fairly close to a 2.18 release

What would you define as the 'minimum changes' needed to get you back onto the current release cycle of AnkiDroid in 2.18 (with the expectation that we continue to improve this feature)

**Comment by BrayanDSO** [comment]
![image](https://github.com/ankidroid/Anki-Android/assets/69634269/c4fabeef-5004-4227-84e6-13cd43ab3f49)

A floating horizontal recorder like iOS's Dynamic islands can fit the idea well

**Comment by promike-github** [comment]
> @promike-github This may take a little while to get off the ground, and we're fairly close to a 2.18 release
> 
> What would you define as the 'minimum changes' needed to get you back onto the current release cycle of AnkiDroid in 2.18 (with the expectation that we continue to improve this feature)

That's ok. Changes take time, and there's no rush from my end.
The most significant issue for me is that a lot more information gets pushed out of view by the visual feedback, and I have to separately approve the recording to listen to it.

I believe these are just temporary files, and in the past, if I hadn't gotten it right the first time, I could always re-record it. The additional step made me inclined to skip the pronunciation part as it felt a bit cumbersome.

If visual feedback is necessary, @BrayanDSO 's suggestion appears elegant and unobtrusive enough not to disrupt the study flow.

I flip through 500-700 cards every day. Even if it just takes a second to hit the extra save button, it adds up to losing 10+ minutes daily, which could mean losing hours each week. The old design wasn't too complicated, but I found it worked better for **me**.

**Comment by MarAdriatne96** [comment]
Hello first of all Thank you for your work. 
Unfortunately, since the last updates of AnkiDroid (around Mar 22), with the new style and options for recording when the user is reviewing flashcards, I’ve been experiencing that after any number of recordings to check whether my answers are correct or not, suddenly I can’t record my voice anymore unless I exit the deck and enter it again (sometimes multiple times) or even exiting the app completely, closing it and open the app again. Sometimes, if I just move to the next card, the problem is “fixed,” but this solution doesn’t always work. It’s like the recording is frozen because none of the buttons work when this happen.

Also, I think it occupied a large portion of the screen. As I read in a previous comment it would be nice if the user could choose either the previous appearance or the new one, because I think some users may found useful the new feature but It's not my case. 

Thank you for your time!

**Comment by criticalAY** [comment]
> since the last updates of AnkiDroid (around Mar 22), with the new style and options for recording when the user is reviewing flashcards,

Did it work fine for 2.17.0? as this behavior is unlikely to happen and shouldn't happen 

> Also, I think it occupies a large portion of the screen. As I read in a previous comment it would be nice if the recorder were like when you record a voice note in WhatsApp for example, and with the option to hear it when the user wants and not an automatic play.

Suggestion noted

**Comment by MarAdriatne96** [comment]
>> Did it work fine for 2.17.0? as this behavior is unlikely to happen and shouldn't happen
>
> This is the version I have (maybe it's my device but as I said This doesn't happened with the other version) :
AnkiDroid Version = 2.17.6 (568538896be943818ca793dd2635822265712def)

Backend Version = 0.1.34-anki23.12.1 (23.12.1 1a1d4d5419c6b57ef3baf99c9d2d9cf85d36ae0a)

Android Version = 9 (SDK 28)

ProductFlavor = play

Manufacturer = motorola

Model = moto e6s

Hardware = mt6765

Webview User Agent = Mozilla/5.0 (Linux; Android 9; moto e6s Build/POBS29.288-60-6-1-29; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/123.0.6312.80 Mobile Safari/537.36

ACRA UUID = aeee54ab-aca6-4feb-b762-90e0169d06ed

Crash Reports Enabled = true

**Comment by david-allison** [comment]
I've been working with @criticalAY on a redesign and this is what we've come up with:

![image](https://github.com/ankidroid/Anki-Android/assets/62114487/0221580d-a524-41c9-900a-9a816af77d6b)

@promike-github @MarAdriatne96 could I have your initial thoughts?

----

Functionality changes will also occur:

* The primary button will be: 'record' -> 'stop' -> 'reply'. A user will no longer be able to stop and start the same recording
* 'Hold to record' will also be implemented
  * Mostly because @promike-github was exceptionally polite & constructive with feedback, especially given our changes in 2.17 were an obvious frustration. THANK YOU

**Comment by david-allison** [comment]
Logic of the recorder currently:

```mermaid
stateDiagram-v2
    [*] --> Empty

    Record --> Playback :💾
    Record --> Empty :🚮
    Playback --> Empty :🚮
    Empty --> Record :⏺️


    state Record {
        r1: Recording
        r2: Paused

        r1 --> r2 : ⏸️
        r2 --> r1 : ⏺️
    }

    state Playback {
        p1: Paused
        p2: Playing

        p1 --> p2 : ▶️
        p2 --> p1 : ⏸️
    }





```


Logic of the new screen

```mermaid
stateDiagram-v2
    [*] --> Empty

    Empty --> Recording : ⏺️
    Recording --> Empty : ❎
    Recording --> Playback : ⏹️
    Playback --> Empty : ❎

    state Playback {
        [*]  --> Paused
        p1: Paused
        p2: Playing

        p1 --> p2 : ▶️
        p2 --> p1 : ↩️
    }
```

**Comment by MarAdriatne96** [comment]
@david-allison I love the design and the idea!

I suggest that when the user past a card either again, hard, good or easy the recorder should be empty again, so is ready to answer the next card. And when the card is flipped the recording should not disappear so the user can check their answer with the other side the card.

However That's the way I use that tool, I don't know if It would be good to add a button to save that recording so it can be listening  in the next card as well if the user wants.
Another thing I would like to point is the place where is situated the recorder, maybe it's not very confortable if the user have to hold the button (like WhatsApp) to record their voice.

I'm going to add another possibility so you will have more options.  But Idk if it could interfere with other types of cards the users use:

![WhatsApp Image 2024-04-21 at 10 09 56 PM](https://github.com/ankidroid/Anki-Android/assets/167715307/56b0d726-8cec-4201-ae51-5a811dd3c4d7)
![WhatsApp Image 2024-04-21 at 10 05 14 PM](https://github.com/ankidroid/Anki-Android/assets/167715307/b74ef32b-c7ef-4eae-ad8a-e14e4f5473ce)
In this way should be more confortable hold the button during the recording.

**Comment by MarAdriatne96** [comment]
@david-allison Sorry for editing so much my comments.

**Comment by david-allison** [comment]
Thanks!!!

With apologies, the 'hold to record' had some additional review-based feedback, so it's likely delayed to 2.19 (I wanted the majority of the redesign in 2.18, rather than blocked on 'hold to record')

This can be tracked: 

* https://github.com/ankidroid/Anki-Android/issues/16282

I'm going to move further improvements to a new issue (please subscribe), but agreed on all counts

* https://github.com/ankidroid/Anki-Android/issues/16288

We've been discussing a new design for the Reviewer in 2.19, and 'bottom of the screen' will fit nicely I feel. Screenshot below

<details>

![image](https://github.com/ankidroid/Anki-Android/assets/62114487/74ce684a-e6c8-45c6-9bb2-81dceacad8c0)

</details>
